sympy/core/mul.py

```diff
@@ -1159,19 +1164,19 @@ def _eval_is_polar(self):
         return has_polar and \
             all(arg.is_polar or arg.is_positive for arg in self.args)
 
-    def _eval_is_real(self):
+    def _eval_is_extended_real(self):
         return self._eval_real_imag(True)
 
     def _eval_real_imag(self, real):
         zero = False
         t_not_re_im = None
 
         for t in self.args:
-            if t.is_complex is False:
+            if t.is_complex is False and t.is_extended_real is False:
```

Comment:
I thought complex implied finite. extended_real could include oo or -oo.

Reply:
On master and in this PR complex does not imply finite and zoo is considered complex in both cases.

(It probably wouldn't be so hard to build on this PR to make complex -> finite -- actually that was my intention but I wanted to get this merged before further work. Thanks for reviewing!)

As @asmeurer says the additional check is needed to exclude oo and -oo. On master we have
```
In [9]: oo.is_complex
Out[9]: True
```
whereas in this PR that would be False. Here oo and -oo are extended_real but not real. Since real implies complex oo is not complex.

Reply:
Actually that doesn't quite follow as necessary but oo and -oo are deliberately not complex in this PR in any case.

Reply:
We can iterate over this PR to make `complex -> finite` throughout. But then, `zoo` would not be `complex` anymore.

Reply:
Maybe we should have `extended_complex` for `zoo`.